iree/vm/bytecode_dispatch.c

```diff
@@ -494,6 +496,59 @@ iree_status_t iree_vm_bytecode_dispatch(
       offset += kRegSize + 4 + kRegSize + kRegSize + kRegSize;
     });
 
+    DISPATCH_OP(SwitchI32, {
+      // let encoding = [
+      //   VM_EncOpcode<opcode>,
+      //   VM_EncOperand<"index", 0>,
+      //   VM_EncIntAttr<"default_value", 32>,
+      //   VM_EncVariadicOperands<"values">,
+      //   VM_EncResult<"result">,
+      // ];
+      int32_t index = OP_R_I32(0);
+      int32_t default_value = OP_I32(2);
+      const iree_vm_register_list_t* src_reg_list =
+          (const iree_vm_register_list_t*)&bytecode_data[offset + 6];
+      offset += kRegSize + 4 + kRegSize + src_reg_list->size * kRegSize;
+      int32_t new_value = default_value;
+      if (index >= 0 && index < src_reg_list->size) {
+        new_value =
+            regs->i32[src_reg_list->registers[index] & IREE_I32_REGISTER_MASK];
+      }
+      OP_R_I32(0) = new_value;
+      offset += kRegSize;
+    });
+
+    DISPATCH_OP(SwitchRef, {
+      // let encoding = [
+      //   VM_EncOpcode<VM_OPC_SwitchRef>,
+      //   VM_EncOperand<"index", 0>,
+      //   VM_EncTypeOf<"result">,
+      //   VM_EncOperand<"default_value", 1>,
+      //   VM_EncVariadicOperands<"values">,
+      //   VM_EncResult<"result">,
+      // ];
+      int32_t index = OP_R_I32(0);
+      int32_t type_id = OP_I32(2);
+      iree_vm_ref_t* default_value = &OP_R_REF(6);
+      int is_move = OP_R_REF_IS_MOVE(6);
+      const iree_vm_register_list_t* src_reg_list =
+          (const iree_vm_register_list_t*)&bytecode_data[offset + 8];
+      offset +=
```

Comment:
Meta-point not related to this pr: I wish that "offset" were called something to signify that it is a global/part of VM state (i.e. we are advancing the instruction pointer). I mis-read "offset" as a local variable often.